### ISSUE TITLE
Unable to build React-Native app

### ISSUE THREAD

**Issue by leonskim** [vmware-samples/workspace-ONE-SDK-integration-samples#23]
### Describe the bug

**TL;DR: We'd need `ws1-sdk-react-native` to be rebuilt with Swift toolchain 5.7.2 (current Xcode toolchain)**

Hi, I've been trying to run the example app(`IntegrationGuideForReactNative`), but I'm not able to build it. I prepared all by running `yarn install`, and `pod install` in `/ios` then when launching the app (`yarn ios`), I see this message:
```
SwiftEmitModule normal x86_64 Emitting\ module\ for\ ws1_sdk_react_native (in target 'ws1-sdk-react-native' from project 'Pods')
```
I modified `Podfile` to exclude irrelevant archs, but didn't work. Also I've launched a new react-native app and installed `ws1-sdk-react-native`, the result was the same.

Then later I found out that `ws1-sdk-react-native` was compiled with Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51) but Xcode 14.2, the version that I'm currently on and also the latest, was compiled with Swift 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51).

Since there's no binary snapshots for Swift 5.7.1, I've compiled the toolchain from the Swift 5.7.1(https://github.com/apple/swift/releases/tag/swift-5.7.1-RELEASE) source, but the error message is still the same. I assume the 5.7.1.135.3 version is not the same as the release version.

So, I can't verify whether it works or not by downgrading the Swift toolchain, but it'd make more sense we have `ws1-sdk-react-native` built with the latest Xcode environment.

Thanks!

### Reproduction steps

1. Clone the repo
2. `cd IntegrationGuideForReactNative`
3. `yarn install`
4. `cd ios`
5. `pod install`
6. `cd ..`
7. `yarn ios`

### Expected behavior

The example app runs without any build errors

**Comment by leonskim** [comment]
Hi @schitranshu,

**Comment by leonskim** [comment]
Hi @schitranshu, I don't want to bother too much but I'd really appreciate if you could give us any updates.

Since the iOS native SDK and its example work well on the latest Xcode, I personally was going to wrap it with `NativeModules` for my React Native app. However, I'm stuck at figuring out how the interfaces are implemented. I took a look at these documentations:
- https://vdc-repo.vmware.com/vmwb-repository/dcr-public/6bfeb7bb-981d-410a-9da2-0586f66bbde6/db034506-7db8-4bde-aef9-027165b63495/WS1iOSDeveloperGuide.pdf
- https://developer.vmware.com/docs/10039/workspace-one-sdk-for-ios--objective-c--developer-guide (Seems different from the Swift implementation)

The key is `WorkspaceOneSDKBridge.xcframework` which contains all the implementation for corresponding `RCT_EXTERN_METHOD` and I'd love to know how it's formed. If I understand correctly, it's a separated utility framework that works as a React Native interface to the AWSDK without having any business-crucial logic or details. So if you could opensource the bridge framework, I'll be able to build the package as whole before you resolve the issue internally. And possibly open a PR for that.

I'm happy to help anything so let me know!

**Comment by schitranshu** [comment]
Fix is to update the podspec , please refer below code 

`require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '13.0'

target 'example' do
  config = use_native_modules!

  use_react_native!(:path => config[:reactNativePath])

  target 'exampleTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  #  use_flipper!({'Flipper' => '0.75.1'})
end


post_install do |installer|
    installer.generated_projects.each do |project|
        project.targets.each do |target|
            target.build_configurations.each do |config|
                if target.name === "RNReanimated"
                    config.build_settings['CLANG_CXX_LANGUAGE_STANDARD'] = 'c++17'
                end
            end
        end
    end
  $WSRN.post_install(installer)
  react_native_post_install(installer)
  __apply_Xcode_12_5_M1_post_install_workaround(installer)
  installer.pods_project.targets.each do |target|
    if target.respond_to?(:product_type) and target.product_type == "com.apple.product-type.bundle"
      target.build_configurations.each do |config|
          config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
          config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.4'
      end
    end
  end
end
target 'example-tvOS' do
  # Pods for example-tvOS

  target 'example-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end
end`

**Comment by schitranshu** [comment]
Fix is to update the podspec file for the app 
[Podfile.txt](https://github.com/vmware-samples/workspace-ONE-SDK-integration-samples/files/12358701/Podfile.txt)

Let me know if still there are any issues .
Thanks,
Shubham Chitranshu

**Comment by leonskim** [comment]
@schitranshu Thank you so much!

**Comment by yashvirpathak** [comment]
I updated the podfile for the iOS app. I am able to build and run the app locally on emulator.
Created a new branch - "ENE-25-ssms-ios-integration-interim" and pushed code into it.
Changes done in following files - 
1. podfile
2. App.js
3. AppDelegate.m

**Comment by yashvirpathak** [comment]
When I tried to run android app, it gave error. I am able to build the app but facing issue while running it locally on emulator.
Error details - 

Exception occurred while executing 'start':
java.lang.SecurityException: Permission Denial: starting Intent { flg=0x10000000 cmp=com.ssms/.MainActivity } from null (pid=9395, uid=2000) not exported from uid 10182
        at com.android.server.wm.ActivityTaskSupervisor.checkStartAnyActivityPermission(ActivityTaskSupervisor.java:1136)
        at com.android.server.wm.ActivityStarter.executeRequest(ActivityStarter.java:1069)
        at com.android.server.wm.ActivityStarter.execute(ActivityStarter.java:742)
        at com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1288)
        at com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1239)
        at com.android.server.am.ActivityManagerService.startActivityAsUserWithFeature(ActivityManagerService.java:3232)
        at com.android.server.am.ActivityManagerShellCommand.runStartActivity(ActivityManagerShellCommand.java:724)
        at com.android.server.am.ActivityManagerShellCommand.onCommand(ActivityManagerShellCommand.java:233)
        at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
        at android.os.ShellCommand.exec(ShellCommand.java:38)
        at com.android.server.am.ActivityManagerService.onShellCommand(ActivityManagerService.java:9681)
        at android.os.Binder.shellCommand(Binder.java:1068)
        at android.os.Binder.onTransact(Binder.java:888)
        at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:5267)
        at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2720)
        at android.os.Binder.execTransactInternal(Binder.java:1344)
        at android.os.Binder.execTransact(Binder.java:1275)
error Failed to start the app.
Error: Command failed: /Users/pyashvir/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell am start -n com.ssms/com.ssms.MainActivity
    at makeError (/Users/pyashvir/Downloads/Project/ssms-1/node_modules/execa/index.js:174:9)
    at module.exports.sync (/Users/pyashvir/Downloads/Project/ssms-1/node_modules/execa/index.js:338:15)
    at tryLaunchAppOnDevice (/Users/pyashvir/Downloads/Project/ssms-1/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchAppOnDevice.js:58:22)
    at /Users/pyashvir/Downloads/Project/ssms-1/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:119:39
    at Array.forEach (<anonymous>)
    at runOnAllDevices (/Users/pyashvir/Downloads/Project/ssms-1/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:117:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Command.handleAction (/Users/pyashvir/Downloads/Project/ssms-1/node_modules/@react-native-community/cli/build/index.js:142:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
